src/target/linux.rs

```diff
@@ -218,6 +299,168 @@ impl LinuxTarget {
 
         Ok(tasks)
     }
+
+    #[cfg(target_arch = "x86_64")]
+    pub fn set_watchpoint(
+        &mut self,
+        watchpoint: Watchpoint,
+    ) -> Result<usize, Box<dyn std::error::Error>> {
+        #[cfg(target_arch = "x86_64")]
+        {
+            let empty = self.find_empty_watchpoint();
+            if empty.is_none() {
+                return Err(Box::new(WatchpointError::NoEmptyWatchpoint));
+            }
+            let index = empty.unwrap();
+
+            let rw_bits: u64 = match watchpoint.typ {
+                WatchpointType::Execute => 0b00,
+                WatchpointType::Read => 0b11,
+                WatchpointType::ReadWrite => 0b11,
```

Comment:
Should these be different from each other?

Reply:
Actually there is no read hardware breakpoint on x86_64. My plan is to somehow distinguish whether ReadWrite was triggered by read or write.